scripts/archivematica_bags_migration.py

```diff
@@ -0,0 +1,212 @@
+
+    document_count = initial_query["hits"]["total"]["value"]
+    results = helpers.scan(
+        client=elastic_client, index=elastic_index, size=5, query=elastic_query
+    )
+
+    with tqdm(total=document_count, file=sys.stdout) as pbar:
+        for result in results:
+            transform(result)
+            # sys.exit(1)
+            pbar.update(1)
+
+
+def bag_creator(s3_client):
+    def create_bag(result):
+        document = result["_source"]
+
+        id = document["id"]
+        bucket = document["location"]["bucket"]
+        path = document["location"]["path"]
+        version = f"v{document['version']}"
+
+        provider = document["location"]["provider"]
+        assert provider == "amazon-s3"
+
+        path_prefix = f"s3://{bucket}/{path}/{version}"
+
+        working_id = id.replace("/", "_")
+        target_folder = "target"
+        working_folder = f"{target_folder}/{working_id}"
+
+        if not os.path.exists(working_folder):
+            os.makedirs(working_folder)
+
+        prefix_patterns = ["bagit.txt", "bag-info.txt", "manifest-", "tagmanifest-"]
+
+        def _write_fetch_file():
+            def _create_fetch_line(file):
+                return f"{path_prefix}/{file['path']}\t{file['size']}\t{file['name']}\n"
+
+            files = document["files"]
+
+            fetch_file = open(f"{working_folder}/fetch.txt", "w")
+            for file in files:
+                fetch_file.write(_create_fetch_line(file))
+            fetch_file.close()
+
+        def _filter_bag_files(prefix):
+            response = s3_client.list_objects_v2(
+                Bucket=bucket, Prefix=f"{path}/{version}/{prefix}"
+            )
+
+            if "Contents" in response:
+                return [content["Key"] for content in response["Contents"]]
+            else:
+                return []
+
+        def _check_expected_prefixes_exist(file_locations):
+            for prefix in prefix_patterns:
+                found_match = False
+
+                for location in file_locations:
+                    if prefix in location:
+                        found_match = True
+
+                if not found_match:
+                    raise RuntimeError(f"Missing any files matching prefix: {prefix}")
+
+        def _get_bag_files():
+            file_locations = []
+            for prefix in prefix_patterns:
+                file_locations = file_locations + _filter_bag_files(prefix)
+
+            _check_expected_prefixes_exist(file_locations)
+
+            for location in file_locations:
+                filename = location.split("/")[-1]
+                save_path = f"{working_folder}/{filename}"
+                s3_client.download_file(bucket, location, save_path)
+
+        def _compress_bag():
+            shutil.make_archive(working_folder, "gztar", working_folder)
+            shutil.rmtree(working_folder, ignore_errors=True)
+
+        _write_fetch_file()
+        _get_bag_files()
+        # Get Archivematica UUID ???
```

Comment:
If you look in an Archivematica bag, there's a METS file in the top level called `METS.[UUID].xml`. That UUID is the one you want.